lhotse/recipes/aishell2.py

```diff
@@ -0,0 +1,90 @@
+"""
+AISHELL2 (~1000 hours) if available(https://www.aishelltech.com/aishell_2).
+"""
+
+import logging
+import os
+import shutil
+import tarfile
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, Optional, Union
+
+from tqdm.auto import tqdm
+
+from lhotse import validate_recordings_and_supervisions
+from lhotse.audio import Recording, RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike, urlretrieve_progress
+
+
+def prepare_aishell2(
+    corpus_dir: Pathlike, output_dir: Optional[Pathlike] = None
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Returns the manifests which consist of the Recordings and Supervisions
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'recordings' and 'supervisions'.
+    """
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    transcript_path = corpus_dir / "AISHELL-2" / "iOS" / "data" / "trans.txt"
+    transcript_dict = {}
+    with open(transcript_path, "r", encoding="utf-8") as f:
+        for line in f.readlines():
+            idx_transcript = line.split()
+            transcript_dict[idx_transcript[0]] = " ".join(idx_transcript[1:])
+
+    manifests = defaultdict(dict)
+    dataset_parts = ["train"]
+    for part in tqdm(
+        dataset_parts, desc="process aishell2 audio, it needs waste some time."
+    ):
+        logging.info(f"Processing aishell2 {part}")
+        # Generate a mapping: utt_id -> (audio_path, audio_info, speaker, text)
+        recordings = []
+        supervisions = []
+        wav_path = corpus_dir / "AISHELL-2" / "iOS" / "data" / "wav"
+        for audio_path in wav_path.rglob("**/*.wav"):
+            logging.info(f"Processing audio path  {audio_path}")
```

Comment:
Will it print lots of logs?

Reply:
It just shows that the program is running normally, it has taken the worry away.